vertical-pod-autoscaler/pkg/updater/restriction/pods_restriction_factory.go

```diff
@@ -392,3 +325,70 @@ func setUpInformer(kubeClient kube_client.Interface, kind controllerKind) (cache
 	}
 	return informer, nil
 }
+
+type singleGroupStats struct {
+	configured             int
+	pending                int
+	running                int
+	evictionTolerance      int
+	evicted                int
+	inPlaceUpdateOngoing   int // number of pods from last loop that are still in-place updating
+	inPlaceUpdateInitiated int // number of pods from the current loop that have newly requested in-place resize
+}
+
+// isPodDisruptable checks if all pods are running and eviction tolerance is small, we can
+// disrupt the current pod.
+func (s *singleGroupStats) isPodDisruptable() bool {
+	shouldBeAlive := s.configured - s.evictionTolerance
+	actuallyAlive := s.running - (s.evicted + s.inPlaceUpdateInitiated)
+	return actuallyAlive > shouldBeAlive ||
+		(s.configured == s.running && s.evictionTolerance == 0 && s.evicted == 0 && s.inPlaceUpdateInitiated == 0)
+	// we don't want to block pods from being considered for eviction if tolerance is small and some pods are potentially stuck resizing
+}
+
+// isInPlaceUpdating checks whether or not the given pod is currently in the middle of an in-place update
+func isInPlaceUpdating(podToCheck *apiv1.Pod) bool {
+	return podToCheck.Status.Resize != ""
+}
+
+// CanEvictInPlacingPod checks if the pod can be evicted while it is currently in the middle of an in-place update.
+func CanEvictInPlacingPod(pod *apiv1.Pod, singleGroupStats singleGroupStats, lastInPlaceAttemptTimeMap map[string]time.Time, clock clock.Clock) bool {
```

Comment:
Any chance we can move this into the pods_inplace_restriction.go file? That way it is closer to where it is used and this factory can be kept (mostly) agnostic of in-place stuff.

Reply:
It's used in both restriction files for what it's worth, do you still want me to do that?